sdk/keyvault/keyvault-admin/test/internal/logger.spec.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as assert from "assert";
+import { createSandbox, SinonSandbox, SinonSpy } from "sinon";
+import { logger } from "../../src/log";
+import { KeyVaultAccessControlClient, KeyVaultBackupClient } from "../../src";
+import { HttpClient, WebResourceLike, HttpOperationResponse, HttpHeaders } from "@azure/core-http";
+import { ClientSecretCredential } from "@azure/identity";
+import { env } from "@azure/test-utils-recorder";
+
+describe("The keyvault-admin clients logging options should work", () => {
+  const keyVaultUrl = `https://keyVaultName.vault.azure.net`;
+  const headers = new HttpHeaders({
+    "x-ms-keyvault-region": "x-ms-keyvault-region-value",
+    "x-ms-keyvault-network-info": "x-ms-keyvault-network-info-value",
+    "x-ms-keyvault-service-version": "x-ms-keyvault-service-version"
+  });
+
+  function makeHTTPMock(path: string, status = 200): HttpClient {
+    return {
+      async sendRequest(httpRequest: WebResourceLike): Promise<HttpOperationResponse> {
+        return {
+          status,
+          headers,
+          request: httpRequest,
+          parsedBody: {
+            id: `${env.KEYVAULT_URI || keyVaultUrl}${path}`,
+            attributes: {}
+          }
+        };
+      }
+    };
+  }
+
+  let mockHttpClient: HttpClient;
+  let sandbox: SinonSandbox;
+  let spy: SinonSpy<any[], void>;
+  let credential: ClientSecretCredential;
+
+  beforeEach(async () => {
+    credential = await new ClientSecretCredential(
+      env.AZURE_TENANT_ID!,
+      env.AZURE_CLIENT_ID!,
+      env.AZURE_CLIENT_SECRET!
```

Comment:
since the tests are mocked, do we really need to pull real credentials from the env?

Reply:
oh not at all! I didn't even sent them when I tested this (they don't need the env variables). I'll clean this up.